### PR TITLE
Draft for wallpapers call

### PR DESCRIPTION
**Pull request by bitigchi**

### CODE REVIEW

File: content/news/2020-09-28-wallpapers-contest.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
++++
+type = "news"
+title = "Call for Wallpapers"
+author = "bitigchi"
+date = "2020-10-01 13:00:00+03:00"
+tags = ["wallpapers", "contest"]
++++
+
+One of the most common feedback that we receive is that the Haiku needs some nice colourful wallpapers shipping with Haiku releases. Suffice to say, that we agree with this completely. We definitely want to improve the current situation. And we need your help.
```
Comment (humdingerb) [inline content/news/2020-09-28-wallpapers-contest.md:9]: I suppose doing the contest, implies we do agree.

```suggestion
One of the most common feedback that we receive is that Haiku needs some nice colourful wallpapers shipping with its releases. We definitely want to improve the current situation. And we need your help.
```

File: content/news/2020-09-28-wallpapers-contest.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Haiku currently ships with only one (1) wallpaper, and it's not a complete wallpaper per se, it's just the Haiku logo positioned on the screen on an appropriate position according to the screen resolution. To be fair, there is a reason we only use the Haiku logo as the default wallpaper; we want Haiku to be usable on systems that have memory as low as 256 MB, therefore each byte that we can spare counts.
```
Comment (humdingerb) [inline content/news/2020-09-28-wallpapers-contest.md:11]: I think the positioning of the logo isn't even resolution dependend...

```suggestion
Haiku currently ships with only one (1) wallpaper, and it's not a complete wallpaper per se, it's just the Haiku logo positioned at fixed coordinates on the screen. To be fair, there is a reason we only use the Haiku logo as the default wallpaper; we want Haiku to be usable on systems that have memory as low as 256 MB, therefore each byte that we can spare counts.
```

File: content/news/2020-09-28-wallpapers-contest.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+We are looking for artists that can contribute to Haiku with some appealing wallpapers. To be more precise, we are looking for either:
+
+- Vector illustrations
+- Painted illustrations
+
+## Guidelines
+
+Artists are free to submit in any category they please. We hope below guidelines will help you in creating the new Haiku wallpapers:
```
Comment (humdingerb) [inline content/news/2020-09-28-wallpapers-contest.md:20]:
```suggestion
Artists are free to submit in any category they please. We hope the guidelines below will help:
```

File: content/news/2020-09-28-wallpapers-contest.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+- Illustrations should be in at least landscape 5K size.
+- Vector illustrations should include a "lite" version, to be used as the default wallpaper (see the second paragraph). Color scheme of this lite variant should blend in with the standard system background colour.
```
Comment (humdingerb) [inline content/news/2020-09-28-wallpapers-contest.md:23]:
```suggestion
- Vector illustrations should include a "lite" version, to be used as the default wallpaper (see the second paragraph). The colour scheme of this lite variant should blend in with the standard system background colour.
```

Maybe quickly summarize what's in the 2nd paragraph. Do you mean "(as a logo placed on the default background)" ?

File: content/news/2020-09-28-wallpapers-contest.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+- We encourage painted illustrations to use subtle haiku themes, such as leaves, Far-Eastern typography, and literature. Anime-themed drawings are also welcome, but should there be human characters, they should not be the main focus.
+- For vector illustrations, use darker blue tones as the main colour. For painted illustrations, use warmer colours.
```
Comment (humdingerb) [inline content/news/2020-09-28-wallpapers-contest.md:25]: I don't see why a vector graphic shouldn't have warm colours, or bitmaps have dark blue ones. It's weirdly specific... :)

File: content/news/2020-09-28-wallpapers-contest.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+- Doodle versions of the Haiku logo are welcome, in-fact we would love to have a coloured Haiku logo doodle for the default wallpaper image (see the part about the lite version).
+- Entries should include a note that the work is licensed under [CC Attribution-ShareAlike 4.0 International](https://creativecommons.org/licenses/by-sa/4.0/) or a similar license.
+- Artists are encouraged to name their work as they please, otherwise you would be leaving the naming part to the mercy of the Haiku developers.
+
+## Submission deadline
+
+There is no strict deadline for the call, we would like you to take as much time as you need. We are planning to release Beta3 in the first quarter of 2021, we would love to include your work by then. Depending on the number of entries before the release date, a vote might be held at the Haiku forums in each respective category.
```
Comment (humdingerb) [inline content/news/2020-09-28-wallpapers-contest.md:32]: I don't think we should mention a timeframe for Beta3. We are wont to slip those... Maybe just:
"We would love to include your work in the next (beta) release"

What are those mentioned categories, vector and bitmap? If so, why even distinguish those. If not, the categories should be stated explicitely somewhere in the text.

We should also mention the file format for the submissions. We need the raw working files, think Illustrator files, for example for vector graphics, Photoshop files for bitmaps etc. Plus SVG / PNG files?

**Review by humdingerb — COMMENTED** [review]
I'm still not sure how we'll deal with setting a background image according to screen resolution on install...
Or do we just include a folder with backgrounds in all possible resolutions and have the user find and set the right one?
Or do we just keep a "lite" logo that we place at a fixed location, hoping it fits to the user's screen resolution?

File: content/news/2020-09-28-wallpapers-contest.md (hunk 5 above)
Comment (bitigchi) [reply content/news/2020-09-28-wallpapers-contest.md:25]: I just thought we would like to keep the blue tones that we use today. For drawings, it should be easy on the eyes. Moreover, all the Far-Eastern drawings I've encountered had warmer colours and natural elements.

File: content/news/2020-09-28-wallpapers-contest.md (hunk 6 above)
Comment (bitigchi) [reply content/news/2020-09-28-wallpapers-contest.md]: We should require WonderBrush format! :P

File: content/news/2020-09-28-wallpapers-contest.md (hunk 3 above)
Comment (bitigchi) [reply content/news/2020-09-28-wallpapers-contest.md]: Fixed. For some reason commit suggestion got greyed out.

File: content/news/2020-09-28-wallpapers-contest.md (hunk 2 above)
Comment (bitigchi) [reply content/news/2020-09-28-wallpapers-contest.md]: I have noticed that the logo position is different on different systems. Sometimes it's on the centre, sometimes it's close to the bottom. How does it work?

**Comment by bitigchi** [comment]
I think there should be a lite version, maybe a cropped version of the main wallpaper as the main thing.

Otherwise we'd have to put different resolutions of the same file, and implement a switch according to the usable RAM.

File: content/news/2020-09-28-wallpapers-contest.md (hunk 2 above)
Comment (humdingerb) [reply content/news/2020-09-28-wallpapers-contest.md]: Are you sure those systems ran the same resolution? AFAIK, the logo appears at fixed X/Y coordinates, origin being the top left corner of the screen. The higher the resolution the further the logo moves up and left.

File: content/news/2020-09-28-wallpapers-contest.md (hunk 5 above)
Comment (humdingerb) [reply content/news/2020-09-28-wallpapers-contest.md]: I just don't see the distinction made between vector and bitmap. I do see that a logo that's supposed to overlay the standard blue background has to fit in, ofc.

Comment (bitigchi) [reply content/news/2020-09-28-wallpapers-contest.md]: They are two different categories. Bitmap can be anything, but the vector is supposed to be a new default background, expanding on the current theme. That's why we need a vector lite version.

Or we can remove all constraints, let artists do what they please, and implement a file selector according to the RAM and resolution.

**Comment by humdingerb** [comment]
First off, you may want to make the general mailing list aware of this PR and the contest in general. Most people don't watch the website PRs and they may have some input on the matter, not least on the technical side.

Maybe you can clarify in the text the "lite" version you propose. Is this an "artistic variation of the Haiku logo" (to be placed on some coordinates of the standard blue (rgb(51,102,152) screen as it is now?
If so, I'm for just 2 categories: "complete background" and "positionable logo". In any format (vector/bitmap)
Possible 3rd category: "tile background". This would solve the resolution issue, but I have yet to see a tile that doesn't look very 90ish... Though, a relatively large tile that won't have to repeat a lot for larger resolutions may just work.

**Comment by bitigchi** [comment]
Updated!

**Comment by nyteshade** [comment]
Have some folders with common resolutions including ultra wide, provide some SVG wallpapers for unexpected resolutions, always lean in on up to 2-3x resolution for supported screens 2048x1536 for 1024x768 resolutions to account for higher than normal DPI monitors. Remember large resolutions can be hand cropped to fit smaller screen sizes. Have some actual logic for what is applied. 

Code changes recommended for best setup

1. [Nice to have] Option and logic to auto place a photo based on resolution. Not too hard. Allow user hinting to reduce code complexity; allow user to specify preference between stretch, keep aspect, center and tile. 

2. During first boot up, ask user to select resolution. 

3. Check available resolutions, get random photo from said folder or tagged collection. 

4. If no set matches available resolutions, choose a SVG image as these will scale to any resolution. Keep these simple, will be slow on some machines. 

5. Fallback to logo and calculate position based on logo size and screen location.

**Comment by nyteshade** [comment]
If you're looking for good images that are royalty free, check unsplash.com.

**Comment by bitigchi** [comment]
> If you're looking for good images that are royalty free, check unsplash.com.

We are eventually planning to include wallpapers from Unsplash, but it's not in the scope of this event.

**Review by humdingerb — COMMENTED** [review]
Sorry, I was sure I've sent these changes days ago...

How do people submit their work? Just upload it somewhere and link to it from a comment of the blog post, or mail it to us, like with the sound contest?

File: content/news/2020-10-04-wallpaper-contest.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+Artists are free to submit in any category they please. We hope the guidelines below will help:
+
+- Illustrations should be positioned in landscape, and should be at least 5K in size (for bitmap submissions).
+- Logo variations are encouraged to preserve the current theme. Doodle versions are welcome.
+- Using subtle haiku themes, such as leaves, Far-Eastern typography, and literature are encouraged. Anime-themed drawings are also welcome, but should there be human characters, they should not be the main focus.
+- If you want to go full 90's, and submit a version suitable for tiled backgrounds, just let us know :).
+- Entries should include a note that the work is licensed under [CC Attribution-ShareAlike 4.0 International](https://creativecommons.org/licenses/by-sa/4.0/) or a similar license.
+- Artists are encouraged to name their work as they please, otherwise you would be leaving the naming part to the mercy of the Haiku developers.
+- Submitted work should be in `*.svg`, `*.kra` or `*.xcf` format.
```
[the remark this answers is not in the document]
Comment (bitigchi) [reply content/news/2020-10-04-wallpaper-contest.md:29]: I think we should keep it FOSS. Wonderbrush submitters can get extra candy 🖌️.

**Pull request closed without merging**